superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance/GPUCore.cpp

```diff
@@ -0,0 +1,507 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// GPUCore.cpp : This file contains the 'main' function. Program execution begins and ends there.
+#include <algorithm>
+#include <array>
+#include <iostream>
+#include <tchar.h>
+#include <vector>
+
+#include <directml.h>
+
+#include "GPUCore.h"
+
+#define DEBUG 0
+#define _PRINT_RESULT 0
+
+/**
+ * @brief Setup GPU and start benchmark.
+ */
+void GPUCore::Run() {
+    int m = opts->m;
+    int n = opts->n;
+    int k = opts->k;
+
+    // Setup GPU objects like device and command list.
+    LoadPipeline();
+
+    int loops = opts->num_loops;
+    std::cout << "GPUCoreFLOPs" << std::endl;
+
+    switch (opts->mode_precision) {
+    case Option::F32: {
+        // Prepare input and output data and buffers.
+        PrepareData<float>(opts->m, opts->n, opts->k);
+        // Setup pipeline and compile operator.
+        LoadAssets(opts->m, opts->n, opts->k, DML_TENSOR_DATA_TYPE_FLOAT32);
+        for (int i = 0; i < opts->num_warm_up; ++i) {
+            InitializeExecuteComputeOp<float>(m, n, k);
```

Comment:
Do we init ComputeOp for each iteration?

Reply:
Whether is too heavy to init each time?

Reply:
separated them, only execute each time